src/app/@theme/components/reactive-form/reactive-control/reactive-control.component.ts

```diff
@@ -0,0 +1,112 @@
+import { FormControl } from '@angular/forms';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ReactiveFormControl } from '../reactive-form-structure';
+import { map, startWith } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'ngx-reactive-control',
+  templateUrl: './reactive-control.component.html',
+  styleUrls: ['./reactive-control.component.scss']
+})
+export class ReactiveControlComponent implements OnInit {
+
+  private _data: ReactiveFormControl;
+
+  @Input('data')
+  public get data(): ReactiveFormControl {
+    return this._data;
+  }
+  public set data(value: ReactiveFormControl) {
+    this._data = value;
+    if (value && this.control) {
+      this.buildControl();
+    }
+  }
+
+  @Input('control') control: FormControl;
+
+  @Output() controlChange ?: EventEmitter < FormControl > = new EventEmitter();
+
+  filterList: Observable < any[] > ;
+  selectList: any[];
+
+  constructor() {}
+
+  ngOnInit() {
+    this.buildControl();
+  }
+
+  private buildControl() {
+    if (this.data) {
+      // const [parent, value] = [this.control.parent, this.control.value];
+
+      switch (this.data.type) {
+        case 'autocomplete':
+          this.buildAutocomplete(this.data);
+          break;
+        case 'input':
+          this.buildInput(this.data);
+          break;
+        case 'button':
+          this.buildButton(this.data);
+          break;
+        case 'select':
+          this.buildSelect(this.data);
+          break;
+        default:
+          break;
+      }
+      // this.control.setValue(this.control.value || this.data.defaultValue);
+      this.control.setValidators(this.data.validators);
+      // this.control.setParent(parent);
+      // this.control.setValue(value);
```

Comment:
Evitar dejar código comentado
```suggestion
      this.control.setValidators(this.data.validators);
```
Alternativamente, dejar comentado también (en código) por qué se dejó comentado y/o cómo proceder al respecto